Core/OfficeDevPnP.Core/Utilities/WebParts/IWebPartPostProcessor.cs

```diff
@@ -5,6 +5,6 @@ namespace OfficeDevPnP.Core.Utilities.WebParts
 {
     public interface IWebPartPostProcessor
     {
-        void Process(WebPartDefinition wpDefinition, File webPartPage);
+        void Process(WebPartDefinition wpDefinition, Web web);
```

Comment:
Hi Paolo @PaoloPia 
Sorry for the delay..It seems I found the way to do that without modifying interfaces.   
Instead of touching interfaces, what if we can simply query web part file with right client context.   
The main problem is `((ClientContext)file.Context).Web` sometimes is not equal to the web, where we are providing our web part. In that case we can clone client context with the right web and run xslt process logic. The only change need to be done in `AddWebPart` method below:  
```csharp
private static WebPartDefinition AddWebPart(Web fileWeb, File webPartPage, WebPartEntity webPart, string zoneId, int zoneIndex)
{
	var limitedWebPartManager = webPartPage.GetLimitedWebPartManager(PersonalizationScope.Shared);
	var oWebPartDefinition = limitedWebPartManager.ImportWebPart(webPart.WebPartXml);

	var wpdNew = limitedWebPartManager.AddWebPart(oWebPartDefinition.WebPart, zoneId, zoneIndex);
	webPartPage.Context.Load(wpdNew);
	webPartPage.Context.ExecuteQueryRetry();
	
	var webPartPostProcessor = WebPartPostProcessorFactory.Resolve(webPart.WebPartXml);

	var currentContext = ((ClientContext) webPartPage.Context);
	var contextWeb = currentContext.Web;

	contextWeb.EnsureProperties(w => w.Url, w => w.Id);
	fileWeb.EnsureProperties(w => w.Url, w => w.Id);

	if (contextWeb.Id.Equals(fileWeb.Id))
	{
		webPartPostProcessor.Process(wpdNew, webPartPage);
	}
	else
	{
		using (var context = currentContext.Clone(fileWeb.Url))
		{
			webPartPage.EnsureProperties(f => f.UniqueId);
			var file = context.Web.GetFileById(webPartPage.UniqueId);
			webPartPostProcessor.Process(wpdNew, file);
		}
	}

	return wpdNew;
}
```  
If you aprove, I will make a separate PR and close this one.